client/src/modules/stock/settings/stock-settings.js

```diff
@@ -0,0 +1,119 @@
+angular.module('bhima.controllers')
+  .controller('StockSettingsController', StockSettingsController);
+
+StockSettingsController.$inject = [
+  'StockSettingsService', 'EnterpriseService', 'util', 'NotifyService', 'SessionService',
+  // 'StockService', 'NotifyService', 'uiGridConstants', 'StockModalService', 'LanguageService',
+  // 'GridGroupingService', 'GridStateService', 'GridColumnService', '$state', '$httpParamSerializer',
+  // 'BarcodeService', 'LotService', 'LotsRegistryService', 'moment',
+];
+
+/**
+ * Stock Settings Controller
+ * This module is a for getting/updating the parameters/settings related to Stock
+ */
+function StockSettingsController(
+  StockSettings, Enterprises, util, Notify, Session,
+  // Stock, uiGridConstants, Modal, Languages,
+  // Grouping, GridState, Columns, $state, $httpParamSerializer,
+  // Barcode, LotService, LotsRegistry, moment,
+) {
+  const vm = this;
+
+  vm.enterprise = {};
+  vm.settings = {};
+
+  let $touched = false;
+
+  // bind methods
+  vm.submit = submit;
+
+  // fired on startup
+  function startup() {
+
+    // load enterprises
+    Enterprises.read()
+      .then(enterprises => {
+        // Assume the the enterprise data has been created already
+        vm.enterprise = enterprises[0];
+        const params = { 'enterprise_id': vm.enterprise.id };
+
+        // Now look up (or create) the stock settings
+        StockSettings.read(null, params)
+          .then(settings => {
+            if (settings.length > 0) {
+              vm.settings = settings[0];
+            } else {
+              // No rows in stock_setting table, create one with defaults for this enterprise
+              StockSettings.create(vm.enterprise.id)
+                .then(_ok => {
+                  // Get the data from the newly created stock_setting row
+                  StockSettings.read(null, params)
+                    .then(settings2 => {
+                      Notify.success('SETTINGS.STOCK_SETTING_DATA_CREATED');
+                      vm.settings = settings2[0];
+                    })
+                    .catch(Notify.handleError);
+                });
+            }
+          });
+      })
+      .catch(Notify.handleError);
+  }
+
+  // form submission
+  function submit(form) {
+    if (form.$invalid) {
+      Notify.danger('FORM.ERRORS.HAS_ERRORS');
+      return 0;
+    }
+
+    // make sure only fresh data is sent to the server.
+    if (form.$pristine && !$touched) {
+      Notify.warn('FORM.WARNINGS.NO_CHANGES');
+      return 0;
+    }
+
+    const changes = util.filterFormElements(form, true);
```

Comment:
So ... this is magic.  It's magic even to me.  I don't think you need it.

I'll post more in detail about it later, when I've had a chance to refresh memory on what it does.  But because you are using `$touched` here, I would just remove it.

Reply:
Yeah, I would just use `vm.settings` and send that back to the server to update the settings.

Reply:
This is essentially the code that was in the corresponding function for the Enterprise settings (enterprises.js, lines 107+).   Please point me to an example that does it the way you suggest.  I tried changing this per your suggestion and it broke things.